cloudvolume/frontends/graphene.py

```diff
@@ -92,17 +91,19 @@ def __getitem__(self, slices):
       watershed coloring.
     """
     try:
-      iter(slices[-1])
-      root_ids = list(slices.pop())
+      slices = list(slices)
+      root_ids = list(toiter(slices.pop()))
       return self.download(
         slices, mip=self.mip, 
         parallel=self.parallel, root_ids=root_ids
       )
     except TypeError: 
       # The end of the array was not iterable, 
       # and thus not the root ids.
+      print('warning no root ids selected, downloading supervoxels')
```

Comment:
Not sure this except clause serves any purpose. It was guarding the access of [-1] previously, but that is no longer true. We are now allowing the last item to be an int instead of a list as well.

Reply:
What happens if someone only passes 3 slices?

Reply:
i tried an alternative approach.. so now it acts like precomputed giving  you straight supervoxels if you slice it normally.. and you pass rootIDs to get remapping behavior.

Reply:
If someone passes fewer than four slices, the rest are filled with implied `:`. The behavior you described seems sensible, though if someone passes in an programatically generated empty list they will be surprised by the result.

Reply:
what do you think of the new option?

Reply:
which tests for len(slices)==4 and does different things depending

Reply:
Hmm, I feel like that makes the interface a little inconvenient. 

I think you'll probably want the following interface?

```python
ws_img = cv[:]    # watershed
ws_img = cv[:,:]  # watershed
ws_img = cv[:,:,:] # still watershed

mapped = cv[:, []] # maps to black
mapped = cv[:, 0] # maps to black
mapped = cv[:,:, [ 5 ]] # agglomerates 5, else black
```

Reply:
I'm not sure that follows normal numpy conventions.    I think if you want to index the 4th axis, you always need to put it in the 4th spot.  Specifying fewer slices than axis is fine, and you default to 'all' of them, but cv[:, []] would be an empty array, cv[:, 0]  would be the all rows, first column, all z, supervoxels. cv[:,:, [ 5 ]]  is z=5. as would mapped = cv[:,:, 5].

Reply:
I guess I wasn't seeing the root ids as a distinct axis but as something else tacked on the end. It does make sense to see it as a kind of axis. It does mean you lose some of these implicit niceties. I'm okay with it if you are.

Reply:
i do think a reasonable option would be to say.. no the __getitem__ interface only gives you slicing into the precomputed format and the download method has the root_ids option.  It's true that there won't be code that easily translates between graphene and precomputed anyway, and you could give precomputed a download method too that makes it behave the same way when you pass rootIds and mask_base=True.  What do you think?

Reply:
I actually quite like this idea to make `__getitem__` behave in a more expected way. The addition of non-slices to it is a kludge that we came up with before the download method existed. I'm not sure it makes sense to backport it to Precomputed. The masking operation is essential to usage of Graphene, but not so much for Precomputed (also the terminology of root_ids doesn't make sense there). It's easy enough to use the `fastremap` module to perform these operations in an explicit way.

If there's a clear use case for backporting I could be convinced otherwise though. I want to keep the interfaces simple, but not so simple they impede people.

Reply:
then i think that's also an argument for making the default behavior of graphene to leave the supervoxel IDs in place, because you'd like downstream analysis code to work reasonably similarly on both sources.  Having black masks all around is very different than be surrounding by other IDs.  Especially considering that id=0 is a special case that means . something different.. i.e. no segmentation.